### ISSUE TITLE
Security Valid 'Set-Cookie' header

### ISSUE THREAD

**Issue by JamoCA** [webhintio/hint#2641]
I received a notification from the Chrome Extension that the set-cookie date is invalid, but there's no link for "more info" in order for me to learn more.  (I also couldn't copy-and-paste the error message. I reported via Twitter and apparently this will be fixed soon.)

This StackOverflow post indicates that "a date in a UTC/GMT format is required by cookies".
https://stackoverflow.com/a/11138591/693068

This hint wasn't reported until after manually fixing both "httponly" and "secure" hints.  Shouldn't this hint always be displayed if the date format is invalid regardless of the "httponly" & "secure" status?

The WebHint version should be displayed in the output so it can be reported if necessary.  (If I manage extensions, I see that it is v0.0.4.)

Thanks.

**Comment by antross** [comment]
@JamoCA Thanks for the report!

> there's no link for "more info" in order for me to learn more

This may be a UX problem on our part. I think you're talking about our link to "[Learn why this is important and how to fix it](https://webhint.io/docs/user-guide/hints/hint-validate-set-cookie-header/)". This is displayed once at the top of each subsection (e.g. "Valid 'Set-Cookie' header"), but I can see how it can easily be missed if you're looking at more than a few hints in this section or have scrolled down.

![image](https://user-images.githubusercontent.com/785009/60278493-31068180-98c5-11e9-8bfe-0d3f64fa9b4e.png)

> (I also couldn't copy-and-paste the error message. I reported via Twitter and apparently this will be fixed soon.

Yes, that was fixed by f37f763ea31770ebbd8f732252f2c949411cd7e5 and will go out with `v0.0.5` of the extension. We should be submitting that update today, but it can take up to a week for it to get approved.

> This hint wasn't reported until after manually fixing both "httponly" and "secure" hints. Shouldn't this hint always be displayed if the date format is invalid regardless of the "httponly" & "secure" status?

Good feedback. Currently the hint checks a few different categories of problems and exits early if any particular category fails, but that's not necessarily the optimal behavior as you point out. I'll open a separate issue to consider changing this.

> The WebHint version should be displayed in the output so it can be reported if necessary. (If I manage extensions, I see that it is v0.0.4.)

I'll open a separate issue for this as well. I could see us listing this next to the "Powered by webhint" text displayed at the bottom of the configuration and results pages.

**Comment by antross** [comment]
> > but there's no link for "more info" in order for me to learn more.

> This may be a UX problem on our part. I think you're talking about our link to "Learn why this is important and how to fix it".

@JamoCA Upon review of our documentation I realize you were probably talking about a link to learn more about the `Expires` format as I see that's missing even from our docs. Definitely something we should fix.

**Comment by JamoCA** [comment]
Yes... that's the type of information that I was looking for.  I don't have any control over how cookies are generated via our application platform (ColdFusion + TomCat) and I needed to report it to Adobe... but didn't know how to articulate it since the one uncopiable sentence in the developer console was all the only information available.  I've reported it as a ColdFusion enhancement/bug here:
https://tracker.adobe.com/#/view/CF-4204632

Upon further review, I noticed (by way of contrast and comparison w/GitHub cookie headers) that ColdFusion "expires" HTTP response headers are title-cased as "Expires". I'm not sure the case of header keys are an issue or a problem for any parsers.  (Probably not an issue, but something I noticed.)

**Comment by antross** [comment]
> Upon further review, I noticed (by way of contrast and comparison w/GitHub cookie headers) that ColdFusion "expires" HTTP response headers are title-cased as "Expires". I'm not sure the case of header keys are an issue or a problem for any parsers. (Probably not an issue, but something I noticed.)

Not an issue as far as I'm aware. I know most of our checks with HTTP headers and values tend to be case-insensitive (we lowercase them before comparing).

**Comment by Amool-kk** [comment]
```Set-Cookie: (cookie-name)=(cookie-value); Expires=(date)```
```Set-Cookie: (cookie-name)=(cookie-value); Max-Age=(number)```

```(cookie-name)=(cookie-value)```
Defines the cookie name and its value. A cookie definition begins with a name-value pair.

```Expires=(date)```
Indicates the maximum lifetime of the cookie as an HTTP-date timestamp.
Date will be in this format : ```Date: (day-name), (day) (month) (year) (hour):(minute):(second) GMT```

If unspecified, the cookie becomes a session cookie. A session finishes when the client shuts down, after which the session cookie is removed. When an Expires date is set, the deadline is relative to the client on which the cookie is being set, not the server. 
Permanent cookies are removed at a specific date (Expires) or after a particular length of time (Max-Age) and not when the client is closed.

Max-Age: Indicates the number of seconds until the cookie expires. A zero or negative number will expire the cookie immediately.

Example : 
```Set-Cookie: id=a3fWa; Expires=Wed, 21 Oct 2015 07:28:00 GMT```
```Set-Cookie: id=a3fWa; Max-Age=2592000```
